uniffi/src/bindings/kotlin/mod.rs

```diff
@@ -69,40 +59,32 @@ pub fn compile_kotlin_bindings(ci: &ComponentInterface, out_dir: &OsStr) -> Resu
 // Execute the specifed kotlin script, with classpath based on the generated
 // artifacts in the given output directory.
 
-pub fn run_kotlin_script(out_dir: Option<&OsStr>, script_file: Option<&str>) -> Result<()> {
+pub fn run_script(out_dir: Option<&Path>, script_file: Option<&Path>) -> Result<()> {
     let mut classpath = std::env::var_os("CLASSPATH").unwrap_or_else(|| OsString::from(""));
     // This lets java find the compiled library for the rust component.
     if let Some(out_dir) = out_dir {
         classpath.push(":");
         classpath.push(out_dir);
         // This lets java use any generate .jar files containing bindings for the rust component.
-        for entry in PathBuf::from(out_dir)
-            .read_dir()
-            .map_err(|_| anyhow!("failed to read directory {:?}", out_dir))?
-        {
-            if let Ok(entry) = entry {
-                if let Some(ext) = entry.path().extension() {
-                    if ext == "jar" {
-                        classpath.push(":");
-                        classpath.push(entry.path().to_str().unwrap());
-                    }
+        for entry in PathBuf::from(out_dir).read_dir()? {
+            let entry = entry?;
+            if let Some(ext) = entry.path().extension() {
+                if ext == "jar" {
+                    classpath.push(":");
+                    classpath.push(entry.path().to_str().unwrap());
```

Comment:
I think `entry.path().as_os_str()` will give you an `OsStr` that you can append without converting and unwrapping.

Reply:
Thanks; turns out plain old `entry.path()` also works